casper/src/main/scala/coop/rchain/casper/dag/BlockDagKeyValueStorage.scala

```diff
@@ -327,4 +363,62 @@ object BlockDagKeyValueStorage {
       stores.deployPool,
       stores.invalidBlocks
     )
+
+  def deployStatus[F[_]: Sync](
+      d: DeployId,
+      deployStore: KeyValueTypedStore[F, DeployId, Signed[DeployData]],
+      deployIndex: KeyValueTypedStore[F, DeployId, BlockHash],
+      invalidBlocksIndex: KeyValueTypedStore[F, DeployId, BlockMetadata],
+      finalizedSet: Set[BlockHash],
+      executedMap: Map[DeployId, Option[String]],
+      expiredSet: Set[DeployId]
+  ): F[String] = {
+    val pooledF   = deployStore.collect { case (dId, _) => dId }.map(_.contains(d))
+    val expiredF  = Sync[F].delay(expiredSet.contains(d))
+    val includedF = deployIndex.collect { case (dId, _) => dId }.map(_.contains(d))
+    val orphanedF = invalidBlocksIndex.collect { case (dId, _) => dId }.map(_.contains(d))
+    val finalizedF = deployIndex.get1(d).map { hashOpt =>
```

Comment:
This is the same as _includedF_.

Reply:
Not sure wdy mean - it is already `get`, isn't it?

Reply:
I guess in both cases (includedF, finalizedF) you can use similar code
```scala
val includedF = deployIndex.get1(d).map(_.isDefined)
```

```scala
val finalizedF = deployIndex.get1(d).map ...
```

Reply:
I've adjusted in https://github.com/rchain/rchain/pull/3682/commits/41051c4d46479ac1a4a5fda23f7f68102d5f66e2
 this to `val includedF = deployIndex.contains(d)`